### DISCUSSION TITLE
Stepan Dyatkovskiy's issues bunch:

### DISCUSSION THREAD

**Discussion by kaomoneus** [voxeltycoon/issues#632] [General]
## Build version
0.85.1 steam

## Platform
MBP Late 2013, 15 inch, 512GB ssd, GeForce 750M GT 2GB

## Sorry for being out of template...

*This issue quite out of proposed standard, but at first I would like just to make a list with short descriptions of all issues discovered on my machine. It is just good not to keep them locally, right?*

*Most of issues/bugs probably would be a Mac OS specific, though.*

*Of course once I get more time, I'll expand them into separate issues following proposed template. So sorry again for breaking a template.*

*So...*

### Video resolution
*This is generic to most of games*
When I change game resolution I would like to keep menus with maximum resolution. It still might be a problem for really slow machines. But as I hope it will solve low fps on mac machines (4K display) keeping menu items sharp.

It might be achieved if game scene will be rendered to texture first with "game" resolution, and then texture will be combined with game window system which works with "menus" resolution.

### Mac OS specific controls

First thing I saw in tutorial is something about holding right button click, pressing ctrl and so on. It is still applicable to Mac OS, but not a common case.

Usually mac os users able to hold only left button down with some modifiers. (On my machine holding left button is configuring as 3-fingers dragging on tauchpad)

It is good to enhance scrolling, for with tauchpad it is possible to perform horizontal and vertical scrolls. Not sure Unity allows it though. But that would be definitely a thing.

...Meanwhile I'm working with WASD stuff.

#### Game button "Add stop" clicks are only possible with mouse
I just spent 5 mins trying to add stop with tauchpad :-) No way. You can only press button with physical left mount button. "Continue" tutorial button works fine by the way.

Same with "send to garage" button. I suspect there is a special class of buttons which changes regular game mode into "pick a target" mode. And this is achievable by l-button only. 

### Tutorial hints (user errors check)
Just bumped into voxeltycoon/bug-tracker#216 . I built second freight station in wrong place, and stupidly expected "continue" to be enabled. I think there should be errors detection like this. E.g. you built station in wrong place, and so on.

### Teleporting to garage
A feature? Looks like a bug though.

### Roads connect
Roads building is much better than in OpenTTD and TTD!

..But there is some issue with connecting roads like this:
![изображение](https://user-images.githubusercontent.com/857042/115971330-e9042480-a558-11eb-97bc-c02e44a53fe5.png)

When you start connection from road at bottom right and target it to road at top-left, first you see error "can't build". But once you keep moving mouse to the right it recovers and you can build it.

### Path finding issues
* Sometimes cars take non-optimal route (to be demonstrated)
* If I remove road which is used by car it still tries to use it, even if there are walk arounds. 
   ![изображение](https://user-images.githubusercontent.com/857042/116003739-48713b80-a610-11eb-9788-2561e05e2379.png)

### Highlight special building
Currently I stucked into issue, that warehouses look almost same like regular building, and there is no way to highlight them, or get a full list of them.

Ideally there should be some sort of query:
1. You select type of unit (car, train, building, etc).
2. You get list of them
3. You also can see them in mode where you highlight things (everything is white, and things are blue) 

### Cars cloning
Yep, there is a clone button. But wouldn't it be good to make individual "clone" button for each car?

### Send cars to nearest garage
It would be good it you could send car to nearest garage with just one click (without picking some particular one).

### Huge worlds
What was good with TTD and OpenTTD were really huge worlds. I know it's not that trivial when it comes to 3D, but perhaps fog + some rendering tasks adjustment should help.

...or it is all because of cities locking mech?

### Cities unlocking
Why? Any parallel with real world? I think that was really nice in predecessor games when you were able to build anything and anywhere (accept case when your reputation is awful)

**Comment by andrewpey** [comment]
Thank you for collecting the list, but we ask you to split it into different issues (some of them already reported).

And regarding your Video resolution question: there's a Render Quality setting that does exactly what you need.

**Comment by kaomoneus** [comment]
> Thank you for collecting the list, but we ask you to split it into different issues (some of them already reported).
> 
> And regarding your Video resolution question: there's a Render Quality setting that does exactly what you need.

Sure I'll try to split. Unfortunately I can do it only on weekends.

Thanks for rendering quality hint, I'll check!
